torch/_inductor/codegen/wrapper.py

```diff
@@ -513,8 +518,22 @@ def generate(self):
                     ),
                 ):
                     line.codegen(self.wrapper_call, device_cm_stack)
+                elif isinstance(line, list):
+                    if isinstance(line[0], OriginOpInfo):
+                        origin_info_list = line
                 else:
+                    # Inspect the contents of line to see if
+                    # it is a triton kernel.  If so add the record_func
+                    with contextlib.ExitStack() as k_stack:
+                        if origin_info_list:
+                            marker_string = str(origin_info_list)
+                            self.wrapper_call.writeline(
+                                f'with record_function("triton_info: {marker_string}"):'
```

Comment:
fyi, if this is torch.profiler.record_function - we're seeing issues with high overhead from this. It's probably fine if you gate it with a config. Or, I'm also working on https://github.com/pytorch/pytorch/pull/106351 which should have lower overhead (at least when the profiler is not on)

Reply:
Hi @davidberard98 - I just got my code working with your changes to fast record_function.  Here's an example screen shot.  I haven't had much time to work on this lately, but I think it is getting close now.

![image](https://github.com/pytorch/pytorch/assets/37422826/0250c37c-6b4d-435e-b4c5-56acc54c8006)